gdal/swig/python/osgeo/utils/auxiliary/util.py

```diff
@@ -96,13 +96,61 @@ def get_ovr_count(filename_or_ds: PathOrDS):
         return bnd.GetOverviewCount()
 
 
-def get_ovr_idx(filename_or_ds: PathOrDS, ovr_idx: Optional[int]):
+def get_sizes_factors_resolutions(filename_or_ds: PathOrDS, dim: Optional[int]=0):
+    ds = open_ds(filename_or_ds)
+    bnd = ds.GetRasterBand(1)
+    ovr_count = bnd.GetOverviewCount()
+    geo_transform = ds.GetGeoTransform()
+    r0 = geo_transform[1], geo_transform[5]
+    s0 = ds.RasterXSize, ds.RasterYSize
+    f0 = 1, 1
+    sizes = [s0]
+    factors = [f0]
+    resolutions = [r0]
+    for iOverview in range(ovr_count):
+        hOverview = bnd.GetOverview(iOverview)
+        if hOverview is not None:
+            s = hOverview.XSize, hOverview.YSize
+            f = s0[0] / s[0], s0[1] / s[1]
+            r = r0[0] * f[0], r0[1] * f[1]
+            sizes.append(s)
+            factors.append(f)
+            resolutions.append(r)
+    if dim is not None:
+        sizes = [x[dim] for x in sizes]
+        factors = [x[dim] for x in factors]
+        resolutions = [x[dim] for x in resolutions]
+    return sizes, factors, resolutions
+
+
+def get_best_ovr_by_resolutions(requested_res: float, resolutions: Sequence[float]):
+    for ovr, res in enumerate(resolutions):
+        if res > requested_res:
+            return max(0, ovr-1)
+    return len(resolutions)-1
+
+
+def get_ovr_idx(filename_or_ds: PathOrDS, ovr_idx: Optional[Union[int, float]]):
```

Comment:
mixing index and resolution in the same argument looks a bit messy. Can't we have a dedicated ovr_res optional argument, mutually exclusive with ovr_idx ?

Reply:
would probably need some test

Reply:
Thanks for you comments!
I also thought it might be a good idea to have a different parameter for resolution, but then two issues arise:
1. As you said, they are mutually exclusive and we'll need to decide what happens if someone sets both. I try to avoid mutually exclusive parameters if possible.
2. If I use a single parameter then any code that uses this function will automatically support this addition so if ovr_idx was passed any levels up the call stack, using a single variable this eliminates the need to modify call the related calls.
Maybe I would rename it to ovr_idx_or_resolution ? This would break any code that called this function with named argument, but as this function was not released yet I guess it's would be OK.

Reply:
>     1. As you said, they are mutually exclusive and we'll need to decide what happens if someone sets both. I try to avoid mutually exclusive parameters if possible.

documentation + exception if that occurs

Reply:
I believe I found a solution that satisfies all the cases mentioned and also clear.